pkg/utils/jsonutils.go

```diff
@@ -58,3 +61,26 @@ func FlattenSingleValue(key string, m map[string]interface{}, child interface{})
 		m[key] = child
 	}
 }
+
+// GetStringFromJson extracts a string value from JSON data based on the provided keys.
+// It uses the supplied workBuf to avoid unnecessary allocations and resizing.
+// If the value is not a string or if an error occurs, it returns an appropriate error.
+func GetStringFromJson(data []byte, workBuf []byte, keys ...string) ([]byte, error) {
+    v, dataType, _, err := jsonparser.Get(data, keys...)
+    if err != nil {
+        return nil, err
+    }
+
+    if dataType != jsonparser.String {
+        if dataType == jsonparser.Null {
+            return nil, fmt.Errorf("key %s has a null value", strings.Join(keys, ", "))
+		}
+        return nil, fmt.Errorf("expected string value for key(s) %s but got %s", strings.Join(keys, ", "), dataType)
+    }
+
+    // Use ResizeSlice to extend workBuf
+    workBuf = ResizeSlice(workBuf, len(v))
+
+    copy(workBuf, v)
+    return workBuf, nil
```

Comment:
Your comment at the top of this function says
> // It uses the supplied workBuf to avoid unnecessary allocations and resizing.

But here you've extracted the value into `v` without using the `workBuf`, and then you copy it to the `workBuf`, so it looks like there's no performance benefit of having a `workBuf`. Can you change how you extract the value so that `workBuf` is actually useful and improves performance?